Update configuration api (#3410)
dapr/docs: daprdocs/content/en/developing-applications/building-blocks/configuration/howto-manage-configuration.md

```

## Retrieve Configuration Items
### Get configuration items

The following example shows how to Get a saved configuration item using the Dapr Configuration API.

> {{< tabs ".NET" Java Python Go "HTTP API (BASH)" "HTTP API (Powershell)">}}

Reviewer: Need to add Javascript example here

Author: Done

> {{< tabs ".NET" Java Python Go "HTTP API (BASH)" "HTTP API (Powershell)">}}

Reviewer: ```suggestion
{{< tabs ".NET" Java Python Go Javascript "HTTP API (BASH)" "HTTP API (Powershell)">}}
```

The following example shows how to Get a saved configuration item using the Dapr Configuration API.

> {{< tabs ".NET" Java Python Go Javascript "HTTP API (BASH)" "HTTP API (Powershell)">}}

Reviewer: ```suggestion
{{< tabs .NET Java Python Go Javascript "HTTP API (BASH)" "HTTP API (Powershell)">}}
```

{{% /codetab %}}

{{% codetab %}}

```go
package main

import (
	"context"
  "fmt"

	dapr "github.com/dapr/go-sdk/client"
)

func main() {
	ctx := context.Background()
	client, err := dapr.NewClient()
	if err != nil {
		panic(err)
	}
	items, err := client.GetConfigurationItems(ctx, "configstore", ["orderId1","orderId2"])
	if err != nil {
		panic(err)
	}
  for key, item := range items {
    fmt.Printf("get config: key = %s value = %s version = %s",key,(*item).Value, (*item).Version)
  }
}
```

{{% /codetab %}}

{{% codetab %}}

Launch a dapr sidecar:

```bash
dapr run --app-id orderprocessing --dapr-http-port 3601
```

In a separate terminal, get the configuration item saved earlier:

```bash
curl http://localhost:3601/v1.0/configuration/configstore?key=orderId1
```

{{% /codetab %}}

{{% codetab %}}

Launch a Dapr sidecar:

```bash
dapr run --app-id orderprocessing --dapr-http-port 3601
```

In a separate terminal, get the configuration item saved earlier:

```powershell
Invoke-RestMethod -Uri 'http://localhost:3601/v1.0/configuration/configstore?key=orderId1'
```

{{% /codetab %}}

{{< /tabs >}}

> ### Subscribe configuration items

Reviewer: ```suggestion
### Subscribe to configuration item updates
```

> Below are the Code examples that leverage Dapr SDKs to subscribe to keys `[orderId1,orderId2]` using `configstore` store. 

Reviewer: ```suggestion
Below are code examples that leverage SDKs to subscribe to keys `[orderId1, orderId2]` using `configstore` store component. 
```

Reviewer: Also it looks like the code examples are inconsistent. Some of them provide CLI commands for running the example and others do not. Can you check for consistency across each of the language SDK examples?

> {{< tabs "Dotnet Extension" "Dotnet Client" Python Go>}}

Reviewer: Need all the SDKs here, which means adding Java and Javascript examples

This is also the suggested order and naming. I am sure this should be ASP.NET Core rather than .NET extension, yes?
```suggestion
{{< tabs ".NET", "ASP.NET Core", Java, Python, Go, Javascript>}}
```

Also suggest adding HTTP versions for Bash and Powershell as you have for unsubscribe

Reviewer: There are still missing examples for Javascript that need to be added

Author: Done

{{< /tabs >}}

### Subscribe to configuration item updates

Below are code examples that leverage SDKs to subscribe to keys `[orderId1, orderId2]` using `configstore` store component. 

> {{< tabs ".NET", "ASP.NET Core", Java, Python, Go, Javascript>}}

Reviewer: Looks like you have .NET and ASP.NET core in the wrong order in the code below. Can you check. 
Also for all these tabs setting, can you check that there is only spaces between them, unless more than one word and then need quotes

Author: Resolved

> {{< tabs ".NET", "ASP.NET Core", Java, Python, Go, Javascript>}}

Reviewer: ```suggestion
{{< tabs .NET "ASP.NET Core" Java Python Go Javascript>}}
```

Author: Done

        print("Subscription ID is", id, flush=True)
        sleep(20)

executeConfiguration()
```

Run the subscriber app along with Dapr Sidecar

```bash
dapr run --app-id orderprocessing -- python3 OrderProcessingService.py
```

In another terminal, update a key using redis-cli:

```bash
redis-cli -p 6379 MSET orderId1 "201||1" orderId2 "202||1"
```

Verify that the subscriber receives the updates:
```
Subscribed item received key=orderId1 value=201 version=1 metadata={}
Subscribed item received key=orderId2 value=202 version=1 metadata={}
```

{{% /codetab %}}

{{% codetab %}}
```go 
package main

import (
	"context"
  "fmt"
  "time"

	dapr "github.com/dapr/go-sdk/client"
)

func main() {
	ctx := context.Background()
	client, err := dapr.NewClient()
	if err != nil {
		panic(err)
	}
  subscribeID, err := client.SubscribeConfigurationItems(ctx, "configstore", []string{"orderId1", "orderId2"}, func(id string, items map[string]*dapr.ConfigurationItem) {
  for k, v := range items {
    fmt.Printf("get updated config key = %s, value = %s version = %s \n", k, v.Value, v.Version)
  }
  })
	if err != nil {
		panic(err)
	}
	time.Sleep(20*time.Second)
}
```

> Run the subscriber app along with Dapr Sidecar

Reviewer: ```suggestion
Run the subscriber app along with Dapr sidecar
```

```bash
dapr run --app-id orderprocessing -- go run main.go
```

In another terminal, update a key using redis-cli:

```bash
redis-cli -p 6379 MSET orderId1 "201||1" orderId2 "202||1"
```

Verify that the subscriber receives the updates:
```
get updated config key=orderId1 value=201 version=1
get updated config key=orderId2 value=202 version=1
```

{{% /codetab %}}
{{< /tabs >}}

> ### Unsubscribe configuration items

Reviewer: ```suggestion
### Unsubscribe from configuration item updates
```

Reviewer: Same comment here as above. looks like the code examples are inconsistent. Some of them provide CLI commands for running the example and others do not. Can you check for consistency across each of the language SDK examples?

Also need code examples for each SDK

After you've subscribed to watch configuration items, you will receive updates for all of the subscribed keys. To stop receiving updates, you need to explicitly call the unsubscribe API.

Following are the code examples unsubscribing using the subscription ID:

> {{< tabs Python Go "HTTP API (BASH)" "HTTP API (Powershell)">}}

Reviewer: Missing all the SDK (need Dotnet, Java and Javascript) and keep the same order as other sections. 

```suggestion
{{< tabs Dotnet Java Python Go Javascript "HTTP API (BASH)" "HTTP API (Powershell)">}}
```

{{< /tabs >}}

### Unsubscribe from configuration item updates

After you've subscribed to watch configuration items, you will receive updates for all of the subscribed keys. To stop receiving updates, you need to explicitly call the unsubscribe API.

Following are the code examples unsubscribing using the subscription ID:

> {{< tabs Dotnet Java Python Go Javascript "HTTP API (BASH)" "HTTP API (Powershell)">}}

Reviewer: ```suggestion
{{< tabs .NET Java Python Go Javascript "HTTP API (BASH)" "HTTP API (Powershell)">}}
```

Author: Done
